backend/proxy/secure_socks_proxy.go

```diff
@@ -143,13 +150,17 @@ func (p *cfgProxyWrapper) NewSecureSocksProxyContextDialer() (proxy.Dialer, erro
 }
 
 func (p *cfgProxyWrapper) getTLSDialer() (*tls.Dialer, error) {
-	certPool := x509.NewCertPool()
+	// legacy file path support
+	if len(p.opts.ClientCfg.RootCAs) > 0 {
+		return p.getTLSDialerFromFiles()
+	}
 
-	if len(p.opts.ClientCfg.RootCAs) == 0 {
+	if len(p.opts.ClientCfg.RootCAsVals) == 0 {
 		return nil, errors.New("one or more root ca are required")
 	}
 
-	for _, rootCA := range p.opts.ClientCfg.RootCAs {
+	certPool := x509.NewCertPool()
+	for _, rootCA := range p.opts.ClientCfg.RootCAsVals {
```

Comment:
Nit. Looks like you could add file contents or vals to a variable and reuse this loop for both cases instead of introducing a new function that makes almost the same thing.

Reply:
Sure let's keep as is. Up to you